apps/opik-backend/src/main/java/com/comet/opik/domain/PromptService.java

```diff
@@ -284,6 +286,21 @@ public void delete(@NonNull UUID id) {
         });
     }
 
+    @Override
+    public void delete(Set<UUID> ids) {
+        if (ids.isEmpty()) {
+            return;
+        }
+
+        String workspaceId = requestContext.get().getWorkspaceId();
+
+        transactionTemplate.inTransaction(WRITE, BatchDeleteUtils.getHandler(
+                PromptDAO.class,
+                repository -> repository.findByIds(ids, workspaceId),
+                Prompt::id,
+                (repository, idsToDelete) -> repository.delete(idsToDelete, workspaceId)));
+    }
```

Comment:
Why do we need this additional check that ids exist as a first step?
If we use just delete, result will be same, it will only delete existing prompts (same for other entities).
Looks like unnecessary DB call.

Reply:
Changed for all relevant resources